### PR TITLE
[BACKLOG-37696][BACKLOG-37985][PDI-19777] changes:

### PR DESCRIPTION
**Pull request by niehusa**
- Changed Rest.java, add HttpClientManager.java
- Solving compilation errors
- Change pom.xml for dependencies
- Delete StringMessageBodyWriter.java & StringMessageBodyWriterTest.java
- Cleanup commented out code in Rest.java
- packaged jersey 2.x dependencies in rest-plugin/lib directory
- Temporarily commented out one test
- Remove exclusions
- Change useTransitiveDependencies to be 'true'
- Use variable for jersey-apache-connector version, change some dependencies to be provided
- Change/Add copyright header
- Implement test testCallEndpointWithGetVerb() using jersey 2.x
- Addressed comment, removed commeted out properties

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 11m 40s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
plugins/rest/assemblies/plugin,plugins/rest/core
```
---
:ok_hand: All tests passed!
 
Tests run: **30**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8987/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/rest/core/pom.xml [hunk 1]
```diff
@@ -19,19 +19,26 @@
     <platform.version>9.6.0.0-SNAPSHOT</platform.version>
     <commons-lang.version>2.6</commons-lang.version>
     <javax.ws.rs-api.version>2.0</javax.ws.rs-api.version>
-    <jersey-client.version>1.19.1</jersey-client.version>
```
Comment (pdesai16) [inline plugins/rest/core/pom.xml:22]: These commented lines can be removed

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 11m 27s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
plugins/rest/assemblies/plugin,plugins/rest/core
```
---
:ok_hand: All tests passed!
 
Tests run: **30**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8987/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 11m 18s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
plugins/rest/assemblies/plugin,plugins/rest/core
```
---
:ok_hand: All tests passed!
 
Tests run: **31**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8987/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: plugins/rest/core/pom.xml [hunk 2]
```diff
@@ -104,17 +106,96 @@
...
+<!--    <dependency>-->
+<!--      <groupId>com.sun.jersey</groupId>-->
+<!--      <artifactId>jersey-bundle</artifactId>-->
+<!--      <version>${jersey-bundle.version}</version>-->
+<!--      <scope>provided</scope>-->
+<!--      <exclusions>-->
+<!--        <exclusion>-->
+<!--          <groupId>*</groupId>-->
+<!--          <artifactId>*</artifactId>-->
+<!--        </exclusion>-->
```
Comment (peterrinehart) [inline plugins/rest/core/pom.xml:184]: Do we need to keep this commented block or can we get rid of it?

File: plugins/rest/core/src/main/java/org/pentaho/di/util/HttpClientManager.java [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+      }
+    };
+
+    SSLContext sc = SSLContext.getInstance( "SSL" );
+    sc.init( null, trustAllCerts, new java.security.SecureRandom() );
+    return sc;
+  }
+
+  //  public static HostnameVerifier getHostnameVerifier( boolean isDebug, ILogChannel log ) {
+  //    return ( hostname, session ) -> {
```
Comment (peterrinehart) [inline plugins/rest/core/src/main/java/org/pentaho/di/util/HttpClientManager.java:276]: Is this method still needed anymore, or can we get rid of it?

**Review by peterrinehart — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 9m 11s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
plugins/rest/assemblies/plugin,plugins/rest/core
```
---
:ok_hand: All tests passed!
 
Tests run: **31**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8987/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by peterrinehart — APPROVED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 10s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl \
plugins/rest/assemblies/plugin,plugins/rest/core
```
---
:ok_hand: All tests passed!
 
Tests run: **31**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-8987/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Pull request merged**